### ISSUE TITLE
"Operation __group_by not found" prometheus query

### ISSUE THREAD

**Issue by mkg20001** [grafana/grafana#67204]
## What happened:

- Prometheus group by query is not properly explained when group by is used

![image](https://user-images.githubusercontent.com/7735145/234279416-4c7a0288-3dfa-461b-9201-67fbe101b86c.png)

## What you expected to happen:

- Query is explained

## How to reproduce it (as minimally and precisely as possible):

- Use the following query `timestamp(group by (hostname, firmware, nodeid) (online))` or similar

## Environment:
- Grafana version: v9.4.7
- Data source type & version: prometheus 5.0.0
- OS Grafana is installed on: nixos unstable
- User OS & Browser: nixos, chrom 113.0.5672.37
- Grafana plugins: default
- Others:

**Comment by mellieA** [comment]
@mkg20001 group by isn't correct syntax for the query. 

https://prometheus.io/docs/prometheus/latest/querying/examples/#using-functions-operators-etc has some examples of how to use "sum by" or "count by" as a way to aggregate your query results by a label.

**Comment by mkg20001** [comment]
it is correct syntax and the query does work

![image](https://user-images.githubusercontent.com/7735145/234511101-07150fb9-f3c0-4fef-83f2-6b2a5b7595f6.png)

![image](https://user-images.githubusercontent.com/7735145/234511175-06c5cba8-b12d-4368-b41c-baa859ecb010.png)

the usecase here is to get all unique pairs of those labels in a range to display them as a table via a later transform -> labels to fields

basically any aggregate function works but group by does exactly what's needed without any extra computation whose value would've gotten thrown away anyways

**Comment by mellieA** [comment]
@mkg20001 can you share which UI you're using to run that query? I tried it in Grafana yesterday and it crashed the query editor. I couldn't find this use documented in the prometheus docs - that said, I'll move it to the right squad to see if we should add this 😄

**Comment by mkg20001** [comment]
The query is

```
timestamp(group by (hostname, firmware, nodeid) (online))
```

which gets all unique pairs of firmware+device, effectivly resulting in upgrade events (with a group by that aggregates time by min to get the first occurence of that pair)

with a bunch of transforms that ends up being this

![image](https://user-images.githubusercontent.com/7735145/234711635-24276578-7b9a-48f7-9687-b9690c81c796.png)

I'm effectivly using prometheus like influxdb using mostly raw data with transformations at the end

Another example:

```
sort(timestamp(present_over_time(online[1m])))
```

![image](https://user-images.githubusercontent.com/7735145/234710635-74bc26e2-544e-48ec-8dac-7e72c182bf24.png)

(also transformations mostly)

I've noticed the queries behave differently in general when being run over grafana versus being run over the prometheus dashboard. where the grafana query results in basically a flood of raw data the prometheus query results in just the latest timestamps for each group (in the group by query). Not sure if this is a bug or a feature or an entierly different issue... but it seems to work.

the json model for both panels:

```
   {
      "datasource": {
        "type": "prometheus",
        "uid": "h1KjclE4k"
      },
      "description": "",
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "custom": {
            "align": "auto",
            "cellOptions": {
              "type": "auto"
            },
            "filterable": false,
            "inspect": false
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": [
          {
            "matcher": {
              "id": "byName",
              "options": "hostname"
            },
            "properties": [
              {
                "id": "displayName",
                "value": "Name"
              }
            ]
          },
          {
            "matcher": {
              "id": "byName",
              "options": "nodeid"
            },
            "properties": [
              {
                "id": "displayName",
                "value": "(id)"
              }
            ]
          },
          {
            "matcher": {
              "id": "byName",
              "options": "Time (min)"
            },
            "properties": [
              {
                "id": "displayName",
                "value": "Joined on"
              }
            ]
          },
          {
            "matcher": {
              "id": "byName",
              "options": "firmware (lastNotNull)"
            },
            "properties": [
              {
                "id": "displayName",
                "value": "Firmware"
              },
              {
                "id": "custom.filterable",
                "value": true
              }
            ]
          }
        ]
      },
      "gridPos": {
        "h": 8,
        "w": 12,
        "x": 0,
        "y": 27
      },
      "id": 34,
      "options": {
        "footer": {
          "countRows": false,
          "fields": "",
          "reducer": [
            "sum"
          ],
          "show": false
        },
        "frameIndex": 0,
        "showHeader": true,
        "sortBy": [
          {
            "desc": true,
            "displayName": "Time (min)"
          }
        ]
      },
      "pluginVersion": "9.4.7",
      "targets": [
        {
          "datasource": {
            "type": "prometheus",
            "uid": "h1KjclE4k"
          },
          "editorMode": "code",
          "expr": "sort(timestamp(present_over_time(online[1m])))",
          "hide": false,
          "interval": "",
          "legendFormat": "__auto",
          "range": true,
          "refId": "A"
        },
        {
          "datasource": {
            "name": "Expression",
            "type": "__expr__",
            "uid": "__expr__"
          },
          "expression": "A",
          "hide": false,
          "reducer": "min",
          "refId": "B",
          "settings": {
            "mode": "dropNN"
          },
          "type": "reduce"
        }
      ],
      "title": "New nodes",
      "transformations": [
        {
          "id": "labelsToFields",
          "options": {
            "mode": "columns"
          }
        },
        {
          "id": "groupBy",
          "options": {
            "fields": {
              "Time": {
                "aggregations": [
                  "min"
                ],
                "operation": "aggregate"
              },
              "firmware": {
                "aggregations": [
                  "lastNotNull"
                ],
                "operation": "aggregate"
              },
              "hostname": {
                "aggregations": [],
                "operation": "groupby"
              },
              "nodeid": {
                "aggregations": [],
                "operation": "groupby"
              }
            }
          }
        },
        {
          "id": "merge",
          "options": {}
        },
        {
          "id": "calculateField",
          "options": {
            "binary": {
              "left": "Time (min)",
              "reducer": "sum"
            },
            "mode": "binary",
            "reduce": {
              "include": [
                "Time (min)"
              ],
              "reducer": "min"
            }
          }
        },
        {
          "id": "calculateField",
          "options": {
            "alias": "cc",
            "binary": {
              "left": "Time (min)",
              "operator": "-",
              "reducer": "sum",
              "right": "C"
            },
            "mode": "binary",
            "reduce": {
              "include": [
                "Time (min)",
                "Min"
              ],
              "reducer": "diff"
            }
          }
        },
        {
          "id": "filterByValue",
          "options": {
            "filters": [
              {
                "config": {
                  "id": "greater",
                  "options": {
                    "value": 0
                  }
                },
                "fieldName": "Time (min)"
              }
            ],
            "match": "all",
            "type": "include"
          }
        },
        {
          "id": "convertFieldType",
          "options": {
            "conversions": [
              {
                "destinationType": "time",
                "targetField": "Time (min)"
              }
            ],
            "fields": {}
          }
        },
        {
          "id": "sortBy",
          "options": {
            "fields": {},
            "sort": [
              {
                "desc": true,
                "field": "Time (min)"
              }
            ]
          }
        },
        {
          "id": "filterByValue",
          "options": {
            "filters": [
              {
                "config": {
                  "id": "greater",
                  "options": {
                    "value": 0
                  }
                },
                "fieldName": "Time (min)"
              }
            ],
            "match": "all",
            "type": "include"
          }
        }
      ],
      "type": "table"
    },
    {
      "datasource": {
        "type": "prometheus",
        "uid": "h1KjclE4k"
      },
      "description": "",
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "custom": {
            "align": "auto",
            "cellOptions": {
              "type": "auto"
            },
            "inspect": false
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": [
          {
            "matcher": {
              "id": "byName",
              "options": "firmware"
            },
            "properties": [
              {
                "id": "displayName",
                "value": "Firmware upgraded to"
              }
            ]
          },
          {
            "matcher": {
              "id": "byName",
              "options": "hostname"
            },
            "properties": [
              {
                "id": "displayName",
                "value": "Hostname"
              }
            ]
          },
          {
            "matcher": {
              "id": "byName",
              "options": "nodeid"
            },
            "properties": [
              {
                "id": "displayName",
                "value": "(id)"
              }
            ]
          },
          {
            "matcher": {
              "id": "byName",
              "options": "Time (min)"
            },
            "properties": [
              {
                "id": "displayName",
                "value": "Upgrade detected at"
              }
            ]
          }
        ]
      },
      "gridPos": {
        "h": 8,
        "w": 12,
        "x": 12,
        "y": 27
      },
      "id": 48,
      "options": {
        "footer": {
          "countRows": false,
          "fields": "",
          "reducer": [
            "sum"
          ],
          "show": false
        },
        "showHeader": true
      },
      "pluginVersion": "9.4.7",
      "targets": [
        {
          "datasource": {
            "type": "prometheus",
            "uid": "h1KjclE4k"
          },
          "editorMode": "code",
          "expr": "timestamp(group by (hostname, firmware, nodeid) (online))",
          "legendFormat": "__auto",
          "range": true,
          "refId": "A"
        }
      ],
      "title": "Firmware upgrades",
      "transformations": [
        {
          "id": "labelsToFields",
          "options": {}
        },
        {
          "id": "filterByValue",
          "options": {
            "filters": [
              {
                "config": {
                  "id": "notEqual",
                  "options": {
                    "value": "$__from"
                  }
                },
                "fieldName": "Time"
              }
            ],
            "match": "all",
            "type": "include"
          }
        },
        {
          "id": "merge",
          "options": {}
        },
        {
          "id": "groupBy",
          "options": {
            "fields": {
              "Time": {
                "aggregations": [
                  "min"
                ],
                "operation": "aggregate"
              },
              "firmware": {
                "aggregations": [],
                "operation": "groupby"
              },
              "hostname": {
                "aggregations": [],
                "operation": "groupby"
              },
              "nodeid": {
                "aggregations": [],
                "operation": "groupby"
              }
            }
          }
        },
        {
          "id": "convertFieldType",
          "options": {
            "conversions": [
              {
                "destinationType": "time",
                "targetField": "Time (min)"
              }
            ],
            "fields": {}
          }
        },
        {
          "id": "sortBy",
          "options": {
            "fields": {},
            "sort": [
              {
                "desc": true,
                "field": "Time (min)"
              }
            ]
          }
        },
        {
          "id": "filterByValue",
          "options": {
            "filters": [
              {
                "config": {
                  "id": "isNotNull",
                  "options": {}
                },
                "fieldName": "Time (min)"
              },
              {
                "config": {
                  "id": "notEqual",
                  "options": {
                    "value": "undefined"
                  }
                },
                "fieldName": "firmware"
              }
            ],
            "match": "all",
            "type": "include"
          }
        }
      ],
      "type": "table"
    },
```
